sw/host/opentitanlib/src/test_utils/load_sram_program.rs

```diff
@@ -25,21 +28,51 @@ use top_earlgrey::top_earlgrey_memory;
 #[derive(Debug, StructOpt, Clone)]
 pub struct SramProgramParams {
     /// Path to the VMEM file to load.
+    #[structopt(long, conflicts_with = "elf")]
+    pub vmem: Option<PathBuf>,
+
+    /// Path to the ELF file to load.
     #[structopt(long)]
-    pub vmem: PathBuf,
+    pub elf: Option<PathBuf>,
 
     /// Address where to load the VMEM file.
-    #[structopt(long, parse(try_from_str = ParseInt::from_str))]
-    pub load_addr: u32,
+    #[structopt(long, parse(try_from_str = ParseInt::from_str), conflicts_with="elf")]
+    pub load_addr: Option<u32>,
+}
+
+/// Describe a file to load to SRAM.
+#[derive(Debug, Clone)]
+pub enum SramProgramFile {
+    Vmem { path: PathBuf, load_addr: u32 },
+    Elf(PathBuf),
 }
 
 impl SramProgramParams {
+    // Convert the command line parameters into a nicer structure.
+    pub fn get_file(&self) -> SramProgramFile {
+        if let Some(path) = &self.vmem {
+            SramProgramFile::Vmem {
+                path: path.clone(),
+                load_addr: self
+                    .load_addr
+                    .expect("you must provide a load address for a VMEM file"),
```

Comment:
Given that this is user input, this shouldn't be panic, but should return `Result` instead.

Reply:
See the discussion on the PR: this requirement should be encoded as a structopt/clap requirement using `requires` but it triggers a clap bug. Since structopt depends on an old version of clap, it makes no sense to report a bug and we are going to get rid of structopt soon anyways. It seems to me that it makes more sense to re-add the proper `requires` when we update clap rather than restructure the flow around an obsolete clap bug.

Reply:
Ok, will do this as part my structopt migration PR